Chain reorganization handling

darosior opened the pull request:
This finally implements our reorganization handling. Like in revaultd, upon noticing a tip change that indicates a reorg happened in our Bitcoin backend we rollback our state to the common ancestor between our state and the new chain, then start rescanning from there. The logic is much more straightforward than in revaultd though, as there is no presigned transactions to care about.

The PR grew a bit large as this needed a bit of preparatory work in order to be reasonably tested (and i noticed a few bugs and cleanups that slipped through review in #29). Please let me know if reviewers prefer that i split the prep work on the commands in another PR.

Fixes #15.

darosior commented:
@edouardparis can you think of an edge case that i didn't properly test here?

darosior commented:
Fixed the failure to compile on `1.48`.

edouardparis reviewed (approved):
utACK 7ca8e613488a3504f2a802aae066965ceb624c59

darosior commented:
Thanks for the review.

I had forgotten to update the CI config to run the new `test_chain.py`. I added another test for completeness sake, as well as cleaning up the existing one a bit. I added a drive-by commit to fix 2 typos i found in the JSONRPC error messages. I did some stress testing and it's holding well, just a spurious failure in an unrelated test (#66). I'll go ahead and merge this if CI is happy.

darosior commented:
ACK a9b0e5e559b0ddddbd8e612771a95182f131e0f0
